contracts/contracts/libraries/BytesLib.sol

```diff
@@ -0,0 +1,498 @@
+// SPDX-License-Identifier: Unlicense
```

Comment:
Might want to do a quick sanity check with legal that there are no implications to including this unlicensed code. Defer to @phutchins on that question.

Reply:
Actually @mds1 it looks like there's also an npm package here, so licensing question aside, why check in this dependency? https://www.npmjs.com/package/solidity-bytes-utils

Reply:
Did not realize there was an npm package, thanks! Updated in beaf5ea

Reply:
Oops forgot to address the license issue. This is [The Unlicense](https://spdx.org/licenses/Unlicense.html) (which is very different from "Unlicensed"), which pretty much lets you do anything, so I don't think there should be any issues here

Reply:
Terrible name, but I agree now that I understand what it is!